source/braille.lua

```diff
@@ -0,0 +1,97 @@
+-- this adds support for love & luajit (via luarocks install utf8)
+if not utf8 then
+  utf8 = require "utf8"
+end
+
+local requireRel
+if arg and arg[0] then
+    package.path = arg[0]:match("(.-)[^\\/]+$") .. "?.lua;" .. package.path
+    requireRel = require
+elseif ... then
+    local d = (...):match("(.-)[^%.]+$")
+    function requireRel(module) return require(d .. module) end
+end
+
+local Maze = requireRel("maze.init")
+
+-- output a maze as braille characters
+-- based on ideas from drawille: https://github.com/asciimoo/drawille
+
+--[[
+http://www.alanwood.net/unicode/braille_patterns.html
+dots:
+   ,___,
+   |1 4|
+   |2 5|
+   |3 6|
+   |7 8|
+   '''''
+or in binary-flags:
+{0x01, 0x08},
+{0x02, 0x10},
+{0x04, 0x20},
+{0x40, 0x80}
+so in order to get walls, it's 2 chars per grid-square
+N + S:
+   ,___,   ,___,
+   |1 4|   |1 4|
+   |   |   |   |
+   |   |   |   |
+   |7 8|   |7 8|
+   '''''   '''''
+add 0x2800 (braille base) like this:
+NE is
+0x2800 + 0x01 + 0x08
+0x2800 + 0x01 + 0x08 + 0x10 + 0x20 + 0x80
+⠉⢹
+]]
+
+function setbits(x, pa)
```

Comment:
I read up on bitwise operators in Lua and the situation seems... complicated. So I'm not against doing this fancy bit of math to emulate true bitwise op, but could you add a comment explaining the reasoning behind the inclusion of such a function as well as some implementation details?
I feel like people coming from languages that have proper support for bitwise operations might be confused by this.

Reply:
yeh, that was exactly why I did it. I was trying to combat not having a good cross-lua way to do bitwise. I will add a note.